met-api/src/met_api/models/event_item.py

```diff
@@ -29,3 +31,36 @@ class EventItem(BaseModel):  # pylint: disable=too-few-public-methods, too-many-
     def save_event_items(cls, event_items: list) -> None:
         """Update widgets.."""
         db.session.bulk_save_objects(event_items)
+        
+    
+    @classmethod
+    def update_event_item(cls, event_data: dict) -> Optional[EventItem or DefaultMethodResult]:
+        """Update Event Item."""
+        query = EventItem.query.filter_by(id=event_data.get('id'))
+        update_fields = dict(   
+            description=event_data.get('description', None),
+            location_name=event_data.get('location_name', None),
+            location_address=event_data.get('location_address', None),
+            start_date=event_data.get('start_date', None),
+            end_date=event_data.get('end_date', None),
+            url=event_data.get('url', None),
+            url_label=event_data.get('url_label', None),
+            sort_index=event_data.get('sort_index', None),
+            widget_events_id=event_data.get('widget_events_id', None),
+            updated_date=datetime.utcnow(),
+        )
+        if not query:
+            return DefaultMethodResult(False, 'Event Not Found', event_data.get('widgets_event_id'))
```

Comment:
I think we have moved away from DefaultMethodResult. If it's still used in the codebase flag it in the group so we create a task to remove them

Reply:
where would you like me to flag it

Reply:
If it's a lot, then we can just create a task to find and remove them, maybe there's one already?